pkg/syndesis/action/startup.go

```diff
@@ -0,0 +1,58 @@
+package action
+
+import (
+	"github.com/openshift/api/apps/v1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Waits for all pods to startup, then mark Syndesis as "Running"
+
+type Startup struct {}
+
+
+func (a *Startup) CanExecute(syndesis *v1alpha1.Syndesis) bool {
+	return syndesisInstallationStatusIs(syndesis, v1alpha1.SyndesisInstallationStatusStarting)
+}
+
+func (a *Startup) Execute(syndesis *v1alpha1.Syndesis) error {
```

Comment:
Isn't this event handler only called once when the status changes from 'Installing' to 'Starting'. When the DCs are not ready then (very likely), when would be this method called be next ? Only when something other changes in the `Syndesis` resource, so I suspect that the status `Installed` is not easily reached. 

I think, instead of listening on `Syndesis` resources, we should listen on `DeploymentConfigs` directly. When such an even comes in, the handler should do the query on the *other* dcs and then when everything is started, fetch the Syndesis resource and update the status from there.

This would be also working when a DC drops out, so the syndesis status could be set to `not running`.

BTW, I would use something liken `Running` or `CrashLoopBack` (or so, sth similar of the DC status) here  to indidicate the runtime status for the application.

Said that, we could also keep a list of all components with their status directly in the Syndesis status, so one can easily see which component would be in an error state without doing a query over all DCs (just a nice add on I guess)

Reply:
I think listening to the DCs is not necessary. The SDK is using a `resyncPeriod` of 5 seconds, so that even if we miss a change event, we are asked later to resync. For example, the first time I create the service account, tokens are usually not available immediately but I need to wait 1-2 resync periods before they are available and I can use them in the template (I was thinking to split the "Installing" phase in two subphases to have a better handling, btw it currently works).

So, unless we want immediate notification when Syndesis is ready, we can wait for the resync trigger.

Tracking the status can be done on the Syndesis status too. It may also be another action that is executed no matter what is the current "InstallationStatus".

Reply:
So the handler is called for every re-sync, so every 5 seconds, even when there has been no change ? If so, I'm fine with that, but then we should include a state like 'CrashLoopback' or sth. when going from 'Running' to broken, i.e. when there are less than expected DCs scaled to 1.